### PR TITLE
Added simulation in bubble sort

### PR DESCRIPTION
**Pull request by lg970325**
I added simulation feature in bubble sort by adding  a new parameter 'simulation'. For example, if the input array is [4,2,3,4,6,1] and you set simulation = True when calling the function, you will see:
[4,2,3,4,6,1]
[2,4,3,4,6,1]
[2,3,4,4,6,1]
[2,3,4,1,4,6]
[2,3,1,4,4,6]
[2,1,3,4,4,6]
[1,2,3,4,4,6]

### CODE REVIEW

**Comment by lg970325** [comment]
Can anyone explain why this makes error? It goes very well in my environment...

File: algorithms/sort/bubble_sort.py [hunk 1]
```diff
@@ -4,19 +4,28 @@
...
 Worst-case performance: O(N^2)
 
+If you call bubble_sort(arr,True), you can see the process of the sort
+Default is simulation = False
+
 """
 
 
-def bubble_sort(arr):
+def bubble_sort(arr,simulation=False):
```
Comment (goswami-rahul) [inline algorithms/sort/bubble_sort.py]: put a space after `, ` there.

File: algorithms/sort/bubble_sort.py [hunk 2]
```diff
@@ -4,19 +4,28 @@
...
     def swap(i, j):
         arr[i], arr[j] = arr[j], arr[i]
 
     n = len(arr)
     swapped = True
+    
+    if simulation == True:
```
Comment (goswami-rahul) [inline algorithms/sort/bubble_sort.py]: just write
```python
if simulation:
    ...
```

File: algorithms/sort/bubble_sort.py [hunk 3]
```diff
@@ -4,19 +4,28 @@
...
+        print(arr)
+    
     while swapped:
         swapped = False
         for i in range(1, n):
             if arr[i - 1] > arr[i]:
                 swap(i - 1, i)
                 swapped = True
+                if simulation == True:
```
Comment (goswami-rahul) [inline algorithms/sort/bubble_sort.py]:
```python
if simulation:
```

**Comment by lg970325** [comment]
Corrected the code:)

**Comment by goswami-rahul** [comment]
@lg970325 The tests passed this time. Weird? opened an issue for this #328

**Comment by keon** [comment]
@goswami-rahul I think so too.
especially Tree/Graph visualization would help a lot:
https://github.com/joowani/binarytree

**Pull request merged**